### PR TITLE
More layout animations

### PR DESCRIPTION
**Pull request by piaskowyk**
## Description

Added more default Layout animations 🚀

## Example 

https://user-images.githubusercontent.com/36106620/120240911-061bc980-c262-11eb-93e9-8f4b5e6edc87.mov

### CODE REVIEW

**Review by Szymon20000 — APPROVED** [review]

File: Example/src/LayoutReanimation/DefaultAnimations.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  ZoomInRight,
+  ZoomInLeft,
+  ZoomInUp,
+  ZoomInDown,
+  ZoomInEasyUp,
+  ZoomInEasyDown,
+  BounceIn,
+  BounceOut,
+  BounceInDown,
+  BounceInUp
```
Comment (jakub-gonet) [inline Example/src/LayoutReanimation/DefaultAnimations.tsx:44]: I wonder if we can group those without any performance penalty, such big import list looks kinda funny 😄

File: Example/src/LayoutReanimation/DefaultAnimations.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+} from 'react-native-reanimated';
+
+const AnimatedBlock = (props: { name: string, animatedStyle: object, defaultShow?: boolean }) => {
+  const { name, animatedStyle, defaultShow } = props;
```
Comment (jakub-gonet) [inline Example/src/LayoutReanimation/DefaultAnimations.tsx:48]: You can create an interface and deconstruct it in the argument list directly without being unreadable.

File: Example/src/LayoutReanimation/DefaultAnimations.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+          <TouchableWithoutFeedback onPress={() => setShow(!show)}>
+            <Animated.View
+              style={styles.animatedBlock}
+              {...animatedStyle}
+            >
+              <Text style={styles.animatedText}>{name}</Text>
+            </Animated.View>
+          </TouchableWithoutFeedback>
+        }
+        {!show &&
```
Comment (jakub-gonet) [inline Example/src/LayoutReanimation/DefaultAnimations.tsx:63]: 🌵  Using ternary instead of `condition` and `!condition` may be more readable.

File: src/reanimated2/animations.ts [hunk 4]
```diff
@@ -340,7 +340,7 @@ export function withStartValue(startValue, animation) {
   });
 }
 
-export function withTiming(toValue, userConfig, callback) {
+export function withTiming(toValue, userConfig?, callback?) {
```
Comment (jakub-gonet) [inline src/reanimated2/animations.ts:343]: Why this change was introduced? This isn't a correct syntax for TS optionals.

File: src/reanimated2/layoutReanimation/defaultAnimations/Flip.ts [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+    const [animation, config] = this.getAnimationAndConfig();
+    const delay = this.delayV;
+
+    return (targetValues) => {
+      'worklet';
+      return {
+        initialValues:{
+          transform: [
+            { perspective: 500 },
+            { rotateY: `0deg`},
```
Comment (jakub-gonet) [inline src/reanimated2/layoutReanimation/defaultAnimations/Flip.ts:303]: 🌵  Apostrophes should work just fine here.

File: src/reanimated2/layoutReanimation/defaultAnimations/Zoom.ts [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    return new ZoomInRotate();
+  }
+
+  build() {
+    const delayFunction = this.getDelayFunction();
+    const [animation, config] = this.getAnimationAndConfig();
+    const delay = this.delayV;
+    const rotate = this.rotateV ? this.rotateV : 0.3;
+
+    return (a) => {
```
Comment (jakub-gonet) [inline src/reanimated2/layoutReanimation/defaultAnimations/Zoom.ts:41]: Do we need this argument?

**Review by jakub-gonet — COMMENTED** [review]
Looks good overall.
